testutil/sims/state_helpers.go

```diff
@@ -42,6 +42,23 @@ func AppStateFn(
 	addressCodec, validatorCodec address.Codec,
 	modules []module.AppModuleSimulation,
 	genesisState map[string]json.RawMessage,
+) simtypes.AppStateFn {
+	return AppStateFnWithExtendedCbs(cdc, addressCodec, validatorCodec, modules, genesisState, nil, nil)
+}
+
+// AppStateFnWithExtendedCbs returns the initial application state using a genesis or the simulation parameters.
+// It panics if the user provides files for both of them.
+// If a file is not given for the genesis or the sim params, it creates a randomized one.
+// genesisState is the default genesis state of the whole app.
+// moduleStateCb is the callback function to access moduleState.
+// rawStateCb is the callback function to extend rawState.
+func AppStateFnWithExtendedCbs(
+	cdc codec.JSONCodec,
+	addressCodec, validatorCodec address.Codec,
+	modules []module.AppModuleSimulation,
+	genesisState map[string]json.RawMessage,
+	moduleStateCb func(moduleName string, genesisState interface{}),
```

Comment:
Don't both `moduleStateCb` and `rawStateCb` all for the same functionality? What is the purpose of including both of them?

The semantics of this function become confused because it is unclear to callers that _first_ the module state callback is executed and then _second_ the raw state callback is executed, meaning things could be overwritten